src/efficient_ui.md

```diff
@@ -1,12 +1,12 @@
 # Efficient UI updates
 
-Relm4 follows the Elm programming model which means that data and widgets are separated. At first glance this might cause a problem. Larger applications need to efficiently update their widgets because rebuilding the whole UI for every update is not an option. But since data and widgets are separated, how do we know which UI elements need to be updated?
+Relm4 follows the Elm programming model which keeps the business logic and its data separated to the widgets. At the first glance you might think this causes problems, but larger applications need this decoupling as it enables them to efficiently modify their UI and business logic with keeping the needed changes local. Rebuilding the whole application with every change of the UI or logic is way too much effort. But since data and UI are separated, how do we know which UI elements need to be updated?
```

Comment:
I think the previous version was easier to understand.

Reply:
updated